client-src/rundump.c

```diff
@@ -172,15 +175,35 @@ main(
 # endif
 #endif
 
-
     /*
      * Build the array
-     */
-
-    g_ptr_array_add(array, g_strdup(dump_program));
-
-    for (i = 1; argv[i]; i++)
-        g_ptr_array_add(array, quote_string(argv[i]));
+     */	 
+	g_ptr_array_add(array, g_strdup(dump_program));
+	
+	// No need to deallocate "prgName", as strrchr is not returning newly allocated memory
+	// but just returning an offset to the string.
+	prgName = g_strrstr(dump_program, "/");
+	if (prgName == NULL)
+	{
+		prgName = dump_program;
+	}
+	else
+	{
+		prgName++;
+	}
+	if (g_strcmp0(prgName, "dump") == 0) {
+		if (validate_dump_option(argc, argv) != 0) {
+			error(" ");
```

Comment:
Why not have the `validate_..` functions call `error(..)` themselves, rather than printing something (which might get lost) and then giving a blank error message (or more oddly, a single space character??) here?  Alternately, `validate_..` could return an allocated error message.

Reply:
Hello Dustin,
Thank you for your review comment.
Yes, I am making changes to move the to error(..) inside the validate_.. functions itself.